test/e2e/scheduling/taint_based_evictions.go

```diff
@@ -0,0 +1,254 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduling
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+)
+
+func newUnreachableNoExecuteTaint() *v1.Taint {
+	return &v1.Taint{
+		Key:    schedulerapi.TaintNodeUnreachable,
+		Effect: v1.TaintEffectNoExecute,
+	}
+}
+
+func runTBEController(cs clientset.Interface, podUpdated, podDeleted chan *v1.Pod, stopCh chan struct{}, ns string) {
+	_, controller := cache.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				obj, err := cs.CoreV1().Pods(ns).List(options)
+				return runtime.Object(obj), err
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return cs.CoreV1().Pods(ns).Watch(options)
+			},
+		},
+		&v1.Pod{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(_, newObj interface{}) {
+				if newPod, ok := newObj.(*v1.Pod); ok {
+					podUpdated <- newPod
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				if delPod, ok := obj.(*v1.Pod); ok {
+					podDeleted <- delPod
+				}
+			},
+		},
+	)
+	go controller.Run(stopCh)
+}
+
+func getTolerationSeconds(tolerations []v1.Toleration) (int64, error) {
+	for _, t := range tolerations {
+		if t.Key == schedulerapi.TaintNodeUnreachable && t.Effect == v1.TaintEffectNoExecute && t.Operator == v1.TolerationOpExists {
+			return *t.TolerationSeconds, nil
+		}
+	}
+	return 0, errors.New("cannot find toleration")
+}
+
+type podTestStruct struct {
+	deleted bool
+	// expectDeleted means if it's expected to be deleted after 30 seoncds
+	expectDeleted           bool
+	tolerationSeconds       int64
+	expectTolerationSeconds int64
+}
+
+var _ = SIGDescribe("TaintBasedEvictions [Serial]", func() {
+	f := framework.NewDefaultFramework("sched-taint-based-evictions")
+	var cs clientset.Interface
+	var ns string
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace.Name
+		// skip if TaintBasedEvictions is not enabled
+		framework.SkipUnlessTaintBasedEvictionsEnabled()
+		// it's required to run on a cluster that has more than 1 node
+		// otherwise node lifecycle manager enters a fully disruption mode
+		framework.SkipUnlessNodeCountIsAtLeast(2)
```

Comment:
hmm... seems this e2e is always skipped b/c gce-serial test is using one-node cluster?

Reply:
confirmed with sig-testing, it's that all "Serial" e2e tests are skipped in CI jobs..

Reply:
FWIW while they're skipped in presubmits, there are definitely some regularly occurring jobs that run Serial tests

Reply:
yeap, I don't want to change current presubmit flow either,